rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md

```diff
@@ -95,5 +97,126 @@ authors:
 :::
 
 ::: good 
-Figure: Good Example {Using arrays helps in listing multiple values under a single key efficiently.} 
+Figure: Good Example of Using arrays helps in listing multiple values under a single key efficiently. 
 :::
+
+
+
+
+#### Use Meaningful Keys
+ 
+The keys you choose for your Frontmatter should be meaningful and descriptive. They act as identifiers for the associated values, so it's essential that they clearly convey the data they represent.
+ 
+- **Descriptive Names:** Instead of using `desc`, use `description`. Instead of `auth`, use `author`.
+- **Consistency:** Stick to a consistent naming convention, whether it's camelCase, snake_case, or kebab-case.
+ 
+Avoid non-descriptive keys:
+ 
+::: greybox
+```
+---
+t: My Article
+auth: John Doe
+---
+```
+:::
+ 
+::: bad 
+Figure: Bad Example of Shortened or unclear keys can lead to confusion.
+:::
+ 
+Opt for clear, meaningful keys:
+ 
+::: greybox
+```
+---
+title: My Article
+author: John Doe
+---
+```
+:::
+ 
+::: good 
+Figure: Good Example of descriptive keys make Frontmatter easy to understand and work with.
+
+
+
+
+#### Explicit Datatypes
+ 
+It's crucial to be explicit about datatypes in Frontmatter. This clarity helps markdown processors understand how to handle the provided metadata correctly.
+ 
+- **Strings vs. Numbers:** If you're representing a year, use a number, e.g., `2023`. If you're mentioning a title or name, use a string, e.g., `"My Article"`.
+- **Booleans:** For binary choices, like true or false, use booleans. For example, `published: true`.
+ 
+Avoid ambiguous datatypes:
+ 
+::: greybox
+```
+---
+year: '2023'
+published: "yes"
+---
+```
+:::
+ 
+::: bad 
+Figure: Bad Example of ambiguous datatypes can lead to parsing errors. 
+:::
+ 
+Be explicit with your datatypes:
+ 
+::: greybox
+```
+---
+year: 2023
+published: true
+---
+```
+:::
+ 
+::: good 
+Figure: Good Example of Explicit datatypes ensure accurate data representation and extraction. 
+:::
+
+
+
+
+#### Avoid Inline HTML
+ 
+While markdown allows the integration of inline HTML, it's recommended to avoid using it within Frontmatter. Using HTML can lead to rendering issues, especially when the markdown is processed by static site generators or other tools.
+ 
+- **Simplicity:** Sticking to markdown syntax within Frontmatter keeps the metadata clean and straightforward. 
+- **Portability:** By avoiding HTML, you ensure that the Frontmatter remains compatible with various markdown processors and platforms.
+ 
+However, some might try to use HTML for additional formatting or structure:
+ 
+::: greybox
+```
+---
+title: <em>My</em> Article
+author: <strong>John Doe</strong>
+---
+```
+:::
+ 
+::: bad 
+Figure: Bad Example of using inline HTML can cause unexpected rendering or parsing issues.
+:::
+ 
+Stick to plain markdown:
+ 
+::: greybox
+```
+---
+title: My Article
+author: John Doe
```

Comment:
this is the same example as the meaningful keys one. I think this should be an example with more detail e.g. more complex properties

add a number, boolean, array, and maybe an object following all the rules

Reply:
Example I have added :
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/96504411/2a4f78b1-ed97-4448-ac94-7ee0d13d1667)